nipoppy_cli/nipoppy/workflows/pipeline.py

```diff
@@ -359,26 +359,23 @@ def run_cleanup(self, **kwargs):
             self.rm(self.dpath_pipeline_work)
         return super().run_cleanup(**kwargs)
 
+    @abstractmethod
     def get_participants_sessions_to_run(
         self, participant: Optional[str], session: Optional[str]
     ):
-        """Return participant-session pairs to run the pipeline on."""
-        # TODO add option in Boutiques descriptor of pipeline
-        # 1. "manifest" (or "all"?)
-        # 2. "downloaded" but not "organized" (from doughnut)
-        # 3. "organized" but not "bidsified" (from doughnut)
-        # 4. "bidsified" but not completed (from doughnut/bagel)
-        # 5. "dataset" (i.e. apply on entire dataset, do not loop over anything)
-
-        # for now just check the participants/sessions that have BIDS data
-        return self.doughnut.get_bidsified_participants_sessions(
-            participant=participant, session=session
-        )
+        """
```

Comment:
Do we need to specify `STEP` here for proper filtering of the bagel? Presumably this is applicable to all bids and proc pipelines?